### ISSUE TITLE
DataBinding generated ModelBuilder interface doesn't include method for newly added data binding variable

### ISSUE THREAD

**Issue by ubuntudroid** [airbnb/epoxy#1007]
After updating to epoxy 4.x when adding a new `variable` to the `data` section of a data binding layout, the generated `BindingModel_` implementation class correctly contains accessors for the new variable. However the according `BindingModel` interface doesn't contain the accessors. And as it's the interface which is accessed from the controller and not the implementation the new property cannot be set through the builder.

Cleaning the project and rebuilding fixes the issue, but that's unfortunately not a viable alternative for the daily development process.

**Comment by elihart** [comment]
It sounds like you're reporting a bug with incremental annotation processing - can you confirm it only happens with incremental builds? If you turn off incremental does the issue go away? Are you using kotlin or java? Does this reproduce 100% of the time when you make this sort of change?

**Comment by ubuntudroid** [comment]
Uh, yeah, I should have included this information into the bug report, my bad. 

I'm using Kotlin, but not incremental builds (as this was prevented by Room at the time of writing the report - this is fixed now on our end). And it did reproduce all the time. But here comes the catch. After downgrading to 3.x (and working with that for a while because I needed to do other things) and then switching back to 4.x it all suddenly worked (incremental and non-incremental). I am not sure this is because of a few project cleans in the meantime - but I'm pretty sure I also tried that out when the initial issue occurred.

I'll keep on working with 4.x now and report back as soon as I see this popping up again. I'll close this ticket for now. Sorry again for the lousy report. 🙏
